src/interval.jl

```diff
@@ -478,6 +478,42 @@ function Base.merge(a::AbstractInterval, b::AbstractInterval)
     return Interval(left, right)
 end
 
+##### ROUNDING #####
+
+for f in (:floor, :ceil, :round)
+    @eval function Base.$f(
+        interval::Interval{T,L,R},
+        args...;
+        on::Type{<:Endpoint},
+    ) where {T,L,R}
+        left = LeftEndpoint(interval)
+        right = RightEndpoint(interval)
+
+        if on === LeftEndpoint
+            # Do nothing if the left endpoint is unbounded
+            if L <: Bounded
+                left = $f(left, args...)
+                # Shifts the interval, span remains the same
+                if R <: Bounded
+                    right = RightEndpoint{T, R}(left.endpoint + span(interval))
+                end
+            end
+        elseif on === RightEndpoint
+            # Do nothing if the right endpoint is unbounded
+            if R <: Bounded
+                right = $f(right, args...)
+                if L <: Bounded
+                    left = RightEndpoint{T, L}(right.endpoint - span(interval))
+                end
+            end
+        else
+            throw(ArgumentError("Unhandled `on` type: $on"))
+        end
+
+        return Interval(left, right)
+    end
+end
```

Comment:
This implementation pre-dates unbounded intervals and I think we can probably write this a little cleaner

Reply:
Possibly you could do the real work in and internal function so `on` can take place in dispatch:
```julia
const RoundingFunctionTypes = Union{typeof(floor), typeof(ceil), typeof(round)}

function Base._round(f::RoundingFunctionTypes, interval::Interval{T,L,R}, on::LeftEndpoint, args...) where {T, L <: Bounded, R <: Bounded}
    left_val = f(first(interval), args...)
    return Interval{T,L,R}(left_val, left_val + span(interval))
end
```

Reply:
If you switch to using `Symbol` for the `on` keyword you can use `Direction{:left}` or `Val{:left}` to still support dispatch